src/components/app/ListOfPublishedProfiles.js

```diff
@@ -0,0 +1,155 @@
+    () =>
+      new Intl.DateTimeFormat(undefined, {
+        month: 'short',
+        day: 'numeric',
+        hour: 'numeric',
+        minute: 'numeric',
+      })
+  ),
+  ancient: memoize(
+    () =>
+      new Intl.DateTimeFormat(undefined, {
+        year: 'numeric',
+        month: 'short',
+        day: 'numeric',
+      })
+  ),
+};
+
+function _formatDate(date: Date, nowTimestamp: Milliseconds) {
+  const timeDifference = nowTimestamp - +date;
+  if (timeDifference < 0 || timeDifference > ONE_YEAR_IN_MS) {
+    return dateFormats.ancient().format(date);
+  }
+  if (timeDifference > ONE_DAY_IN_MS) {
+    return dateFormats.thisYear().format(date);
+  }
+  return dateFormats.thisDay().format(date);
+}
+
+function _formatRange(range: StartEndRange): string {
+  return formatSeconds(range.end - range.start, 3, 1);
+}
+
+type PublishedProfileProps = {|
+  +profileData: ProfileData,
+  +nowTimestamp: Milliseconds,
+|};
+
+function PublishedProfile({
+  profileData,
+  nowTimestamp,
+}: PublishedProfileProps) {
+  let { urlPath } = profileData;
+  if (!urlPath.startsWith('/')) {
+    urlPath = '/' + urlPath;
+  }
+  const location = `${window.location.origin}/${urlPath}`;
+  return (
+    <li className="publishedProfilesListItem">
+      <a className="publishedProfilesLink" href={location}>
+        <div className="publishedProfilesDate">
+          {_formatDate(profileData.publishedDate, nowTimestamp)}
+        </div>
+        <div className="publishedProfilesName">
+          <strong>
+            {profileData.name
+              ? profileData.name
+              : `Profile #${profileData.profileToken.slice(0, 6)}`}
+          </strong>{' '}
+          ({_formatRange(profileData.publishedRange)})
+        </div>
+        <ProfileMetainfoSummary meta={profileData.meta} />
+      </a>
+    </li>
+  );
+}
+
+type State = {|
+  profileDataList: null | ProfileData[],
+|};
+
+export class ListOfPublishedProfiles extends PureComponent<{||}, State> {
+  state = {
+    profileDataList: null,
+  };
+
+  async componentDidMount() {
+    const profileDataList = await listAllProfileData();
+
+    // It isn't ideal to use a setState here, but this is the only way.
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      profileDataList,
+    });
+  }
+
+  render() {
+    const { profileDataList } = this.state;
+
+    if (!profileDataList) {
+      return null;
+    }
+
+    if (!profileDataList.length) {
+      return (
+        <main className="publishedProfiles">
+          <AppHeader />
+          <h2 className="photon-title-30">Uploaded Recordings</h2>
+          <p className="photon-body-30">No profile has been published yet!</p>
+        </main>
+      );
+    }
+
+    const nowTimestamp = Date.now();
+
+    return (
+      <main className="publishedProfiles">
+        <AppHeader />
+        <h2 className="photon-title-30">Uploaded Recordings</h2>
+        {/* TODO Maybe we should use a grid layout later */}
```

Comment:
Flex seems fine to me.

Reply:
I wanted a grid to align things better between the various rows, but we'd need subgrid and this is hard, so I took an easier path which isn't so bad...